runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/MonitoringUtil.java

```diff
@@ -219,6 +202,31 @@ public static String getGcloudCancelCommand(DataflowPipelineOptions options, Str
   }
 
   public static State toState(String stateName) {
-    return MoreObjects.firstNonNull(DATAFLOW_STATE_TO_JOB_STATE.get(stateName), State.UNRECOGNIZED);
+    switch (stateName) {
```

Comment:
I also inlined the map because it was less flexible and less readable than a simple `switch`.